File: src/main/java/nodes/expressions/Parameter.java
package nodes;

import nodes.expressions.Expression;

public class Parameter extends Expression {
    private String name, type, value;

    public Parameter(String name, String type) {
        this.name = name;
        this.type = type;
    }

    public Parameter(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        if (value == null)
            return type + " " + name;
        return value;
    }
}

File: src/main/kotlin/nodes/expressions/Parameter.kt
package nodes.expressions

class Parameter : Expression {
    private var name: String? = null
    private var type: String? = null
    private var value: String? = null

    constructor(name: String?, type: String?) {
        this.name = name
        this.type = type
    }

    constructor(value: String?) {
        this.value = value
    }

    override fun toString(): String {
        return value ?: "$type $name"
    }
}